api/src/main/java/org/apache/brooklyn/api/catalog/CatalogItem.java

```diff
@@ -103,6 +103,11 @@ public static CatalogItemType ofTargetClass(Class<? extends BrooklynObject> type
     @Deprecated
     @Nullable public String getJavaType();
 
+    /**
+     * @return The underlying java type of the item represented or the catalog item ID, described via a YAML spec.
+     */
+    @Nullable public String getReferenceType();
```

Comment:
The difference is that it may refer either to a java type, or to a catalog item, so getJavaType is the wrong name for it, so it makes sense to deprecate getJavaType and replace it with getReferenceType (this also addresses the other comment below)